### DISCUSSION TITLE
StyleGuide e Metodós com React

### DISCUSSION THREAD

**Discussion by Adejair** [frontendbr/forum#131] [Perguntas e Respostas]
Fala galerinha, beleza ?

Gostaria de saber qual styleguide e metódos vocês usam com o **react**. Por exemplo na criação de uma classe com React 

Se acham mais viavel usar classes do es6(utilizando o babel), ou se usam o própio metódo do react para criação e porque utilizam de tal forma.

**Babel**

``` Javascript
class App extends React.component {
   ...
}
```

**React Method**

``` Javascript
var MyAwesomeReactClass = React.createClas({
   ...
});
```

Ou se utilizam da forma de criação de objetos ou se criam da forma do JSX.

**Objects**

``` Javascript
var MyClass = React.createClass({
   render: {
       return (
          <div>
            <p> Utilizando o render de forma de objeto </p>
          </div> 
      )
   }
});
```

**JSX Method**

``` Javascript
var MyClass = React.createClass({
   render() {
      return (
          <div>
             <p> Utilizando o render na forma de JSX </p>
          </div>  
      )
   }
});
```

Ahh, e se vocês usam algum styleguide sem ser o da airbnb pra JSX :+1:

**Comment by diessica** [comment]
Além do style guide do Airbnb, [React Patterns](https://github.com/planningcenter/react-patterns) é bacana de seguir e me orientou bastante.

## 

Sobre sintaxe, existem sim alguns pontos importantes, mas qual escolher vai mais da sua preferência do que qualquer outra coisa!

### `React.Component`
- Não tem autobinding nos métodos – faz você usar `bind(this)` ou [_property initializers_](https://facebook.github.io/react/blog/2015/01/27/react-v0.13.0-beta-1.html#autobinding).
- Não tem mixins – encoraja o uso de _Higher-order Components_.
- Não tem `this.isMounted` – não faz falta.

Se você não quer depender do Babel, `React.createClass` faz um ótimo trabalho. Atualmente eu utilizo apenas `React.Component`.

**Comment by felisio** [comment]
Complementando o link do Style Guide do [Airbnb](https://github.com/airbnb/javascript/tree/master/react)

Lembrando que se vc for usar a arquitetura flux como o redux, nem tudo se aplica..., geralmente eu faço levantamento do que acho mais interessante e aplico no meu projeto.
